geometry-builder.cpp

```diff
@@ -209,11 +209,14 @@ geom_ptr geometry_builder::create_simple_poly(GeometryFactory &gf,
     std::unique_ptr<std::vector<Geometry *> > empty(new std::vector<Geometry *>);
     geom_ptr geom(gf.createPolygon(shell.release(), empty.release()));
 
-    if (!geom->isValid()) {
+    if (geom->isEmpty() || !geom->isValid()) {
```

Comment:
This code block will attempt to create a buffer(0) around an empty geometry, and then throw an exception because the geometry is still empty.